Enabled min, max, minall, maxall, cmin, cmax, cmaxValue, cminValue for bool tensors

izdeby opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #21033 Enable all and any for bool tensors
* #21032 Enabled add, sum, mul and neg for bool tensor
* **#21031 Enabled min, max, minall, maxall, cmin, cmax, cmaxValue, cminValue for bool tensors**
* #21030 Added possibility to index scalars by bool masks

This PR is a part of a stack which will change result tensor type of comparison ops from uint8 to bool. As this change is rather big and a lot of prep work is needed, im breaking it into a stack.

Changes in this PR:
- Enable min, max, minall, maxall, cmin, cmax, cmaxValue, cminValue for bool tensors by moving their declarations and definitions above the `#if !defined(TH_REAL_IS_BOOL)` macro.

Differential Revision: [D15530499](https://our.internmc.facebook.com/intern/diff/D15530499)

izdeby committed:
Enabled min, max, minall, maxall, cmin, cmax, cmaxValue, cminValue for bool tensors

izdeby committed:
Update on "Enabled min, max, minall, maxall, cmin, cmax, cmaxValue, cminValue for bool tensors"

Enabled min, max, minall, maxall, cmin, cmax, cmaxValue, cminValue for bool tensors

gh-metadata: pytorch pytorch 21031 gh/izdeby/2/head

ezyang commented:
Adding on to the documentation question from the previous PR; in general, if I want to find out what the semantics of boolean tensors are in the documentation, how can I find this out? If I look at any given function (like max) and I want to know how it behaves on a boolean tensor, how do I know or not?

izdeby committed:
Update on "Enabled min, max, minall, maxall, cmin, cmax, cmaxValue, cminValue for bool tensors"

Enabled min, max, minall, maxall, cmin, cmax, cmaxValue, cminValue for bool tensors

gh-metadata: pytorch pytorch 21031 gh/izdeby/2/head

izdeby committed:
Update on "Enabled min, max, minall, maxall, cmin, cmax, cmaxValue, cminValue for bool tensors"

Enabled min, max, minall, maxall, cmin, cmax, cmaxValue, cminValue for bool tensors

gh-metadata: pytorch pytorch 21031 gh/izdeby/2/head
